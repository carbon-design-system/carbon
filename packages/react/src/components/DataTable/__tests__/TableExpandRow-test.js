/**
 * Copyright IBM Corp. 2016, 2018
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import { mount } from 'enzyme';
import { Table, TableBody, TableExpandRow, TableExpandedRow } from '../';

describe('DataTable.TableExpandRow', () => {
  let mockProps;

  beforeEach(() => {
    mockProps = {
      className: 'custom-class',
      isExpanded: false,
      onExpand: jest.fn(),
      ariaLabel: 'Aria label',
    };
  });

  it('should render', () => {
    const wrapper = mount(
      <Table>
        <TableBody>
          <TableExpandRow {...mockProps} />
          <TableExpandedRow colSpan={1} />
        </TableBody>
      </Table>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should initially not define `data-previous-value`', () => {
    const wrapper = mount(
      <Table>
        <TableBody>
          <TableExpandRow {...mockProps} />
          <TableExpandedRow colSpan={1} />
        </TableBody>
      </Table>
    );
    expect(
      Object.keys(wrapper.find('TableCell').first().props()).indexOf(
        'data-previous-value'
      ) !== -1
    ).toBe(true);
    expect(
      wrapper.find('TableCell').first().prop('data-previous-value')
    ).not.toBeDefined();
  });

  it('should expand when the expand button is clicked', () => {
    const initialWrapper = mount(
      <Table>
        <TableBody>
          <TableExpandRow {...mockProps} />
          <TableExpandedRow colSpan={1} />
        </TableBody>
      </Table>
    );
    initialWrapper.find('button').simulate('click');
    expect(mockProps.onExpand).toHaveBeenCalledTimes(1);

    const expandedWrapper = mount(
      <Table>
        <TableBody>
          <TableExpandRow {...mockProps} isExpanded />
          <TableExpandedRow colSpan={1} />
        </TableBody>
      </Table>
    );
    expect(
      expandedWrapper.find('TableCell').first().prop('data-previous-value')
    ).toBe('collapsed');
  });
});
