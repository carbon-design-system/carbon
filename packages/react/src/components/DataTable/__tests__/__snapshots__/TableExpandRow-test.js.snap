// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataTable.TableExpandRow should render 1`] = `
<Table
  isSortable={false}
  overflowMenuOnHover={true}
>
  <div
    className="bx--data-table-content"
  >
    <table
      className="bx--data-table bx--data-table--no-border"
    >
      <TableBody
        aria-live="polite"
      >
        <tbody
          aria-live="polite"
        >
          <TableExpandRow
            ariaLabel="Aria label"
            className="custom-class"
            expandHeader="expand"
            isExpanded={false}
            onExpand={[MockFunction]}
          >
            <tr
              className="bx--parent-row custom-class"
              data-parent-row={true}
            >
              <TableCell
                className="bx--table-expand"
                headers="expand"
              >
                <td
                  className="bx--table-expand"
                  headers="expand"
                >
                  <button
                    aria-label="Aria label"
                    className="bx--table-expand__button"
                    onClick={[MockFunction]}
                    type="button"
                  >
                    <ForwardRef(ChevronRight16)
                      className="bx--table-expand__svg"
                    >
                      <Icon
                        className="bx--table-expand__svg"
                        fill="currentColor"
                        height={16}
                        preserveAspectRatio="xMidYMid meet"
                        viewBox="0 0 16 16"
                        width={16}
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <svg
                          aria-hidden={true}
                          className="bx--table-expand__svg"
                          fill="currentColor"
                          focusable="false"
                          height={16}
                          preserveAspectRatio="xMidYMid meet"
                          viewBox="0 0 16 16"
                          width={16}
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M11 8L6 13 5.3 12.3 9.6 8 5.3 3.7 6 3z"
                          />
                        </svg>
                      </Icon>
                    </ForwardRef(ChevronRight16)>
                  </button>
                </td>
              </TableCell>
            </tr>
          </TableExpandRow>
          <TableExpandedRow
            colSpan={1}
          >
            <tr
              className="bx--expandable-row"
              data-child-row={true}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              <TableCell
                colSpan={1}
              >
                <td
                  colSpan={1}
                >
                  <div
                    className="bx--child-row-inner-container"
                  />
                </td>
              </TableCell>
            </tr>
          </TableExpandedRow>
        </tbody>
      </TableBody>
    </table>
  </div>
</Table>
`;
